README.md

```diff
@@ -500,96 +441,113 @@ export default MyComponent = () => {
+  If you had code like this:
 
-#### `authService.handleAuthentication()`
+  ```javascript
+  authService.logout('/goodbye');
+  ```
 
-(async method) Parses tokens from the url and stores them.  Done when handling a redirect back to the application from the issuer.  See the `LoginCallback` component for an example of use.
+  it can be rewritten as:
 
-### `authService.setFromUri(uri)`
+  ```javascript
+  oktaAuth.signOut({ postLogoutRedirectUri: window.location.orign + '/goodbye' });
+  ```
 
-(synchronous method) Store the current URL state before a redirect occurs.  Called internally by `authService.login`.  If a relative path is passed it will be converted to an absolute URI before storage.
+  **Note** that the value for `postLogoutRedirectUri` must be an absolute URL. This URL must also be on the "allowed list" in your Okta app's configuration. If no options are passed or no `postLogoutRedirectUri` is set on the options object, it will redirect to `window.location.origin` after sign out is complete.
 
-#### `authService.getFromUri()`
+- `getAccessToken` and `getIdToken` have been changed to synchronous methods
 
-(synchronous method) Returns the stored URI string stored by `setFromUri` and removes it from storage.  See the `LoginCallback` component for an example of use.  
+  With maintaining in-memory [AuthState][] since [Okta Auth SDK][] version 4.1, token values can be accessed in synchronous manner.
 
-#### `authService.getTokenManager()`
+- `handleAuthentication` is replaced by `handleLoginRedirect`
 
-(async method) Resolves with the internal [TokenManager](https://github.com/okta/okta-auth-js#tokenmanager).
+  `handleLoginRedirect` is called by the `OktaLoginCallback` component as the last step of the login redirect authorization flow. It will obtain and store tokens and then call `restoreOriginalUri` which will return the browser to the `originalUri` which was set before the login redirect flow began.
 
-#### `authService.updateAuthState()`
+- `authState` related methods have been collected in [Okta Auth SDK AuthStateManager](https://github.com/okta/okta-auth-js#authstatemanager)
+  - Change `authService.updateAuthState` to `oktaAuth.authStateManager.updateAuthState`
+  - Change `authService.getAuthState` to `oktaAuth.authStateManager.getAuthState`
+  - Change `on` to `oktaAuth.authStateManager.subscribe`
+  - `clearAuthState`, `emitAuthState` and `emit` have been removed
 
-(async method) Triggers a re-evaluation of whether a user is considered authenticated.  Might be need to be triggered manually for users that pass overrides to `onAuthRequired` to the Security component or to the `Auth` constructor.  Does NOT perform a login, simply re-evaluates the current authenticated status. An 'authStateChange' event is emitted once the re-evaluation is complete. 
+- `isAuthenticated` will be true if **both** accessToken **and** idToken are valid
```

Comment:
suggest: "if and only if"

What if their scopes don't involve both tokens?  Is that still a case that can happen?

Reply:
> What if their scopes don't involve both tokens? Is that still a case that can happen?

`by default` should be added. We may also want to add an explanation in auth-js#readme about how to use the `transformAuthState` callback to handle custom scopes cases.